zest/releaser/baserelease.py

```diff
@@ -57,7 +57,7 @@
 class Basereleaser(object):
 
     def __init__(self, vcs=None):
-        os.environ["ZESTRELEASER"] = "We are called from within zest.releaser"
+        os.environ[str("ZESTRELEASER")] = str("We are called from within zest.releaser")
```

Comment:
What does this fix? I don't think this changes anything in any Python version.
I tried the original in Python 2.7, 3.7, 3.9, pypy2, pypy3: all worked fine.
So simply this:

```
$ python
>>> import os
>>> os.environ["ZESTRELEASER"] = "We are called from within zest.releaser"
```

Reply:
I actually got this stack trace when running `release -v` on my repo:
```Traceback (most recent call last):
  File "C:\Python27-64\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\Python27-64\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Development\sources\stage\_venv\Scripts\release.exe\__main__.py", line 7, in <module>
  File "c:\development\sources\stage\_venv\lib\site-packages\zest\releaser\release.py", line 346, in main
    releaser = Releaser()
  File "c:\development\sources\stage\_venv\lib\site-packages\zest\releaser\release.py", line 63, in __init__
    baserelease.Basereleaser.__init__(self, vcs=vcs)
  File "c:\development\sources\stage\_venv\lib\site-packages\zest\releaser\baserelease.py", line 70, in __init__
    'name': self.vcs.name,
  File "c:\development\sources\stage\_venv\lib\site-packages\zest\releaser\git.py", line 31, in name
    package_name = self.get_setup_py_name()
  File "c:\development\sources\stage\_venv\lib\site-packages\zest\releaser\vcs.py", line 93, in get_setup_py_name
    utils.execute_command(utils.setup_py('egg_info'))
  File "c:\development\sources\stage\_venv\lib\site-packages\zest\releaser\utils.py", line 866, in execute_command
    result = _execute_command(command)
  File "c:\development\sources\stage\_venv\lib\site-packages\zest\releaser\utils.py", line 810, in _execute_command
    process = subprocess.Popen(command, **process_kwargs)
  File "C:\Python27-64\lib\subprocess.py", line 394, in __init__
    errread, errwrite)
  File "C:\Python27-64\lib\subprocess.py", line 644, in _execute_child
    startupinfo)
TypeError: environment can only contain strings```

And I traced that back to having this in the the Zest release as a unicode object in the environment passed to subprocess:
```
{ ...
u'ZESTRELEASER: u'We are called from within zest.releaser'
... }
```

And I figured that was due to a combination of using `unicode_literals` on Python 2.7 (Python 2.7.15, to be precise). It may also be some or another oddity on my installation, because I also did not see that specific problem on another installation.

Anyway, forcing this into 'str' type fixed the issue for me.

Reply:
Ah, I did not realise we are using `unicode_literals`. I don't think I have ever used them, but I see we use them in `zest.releaser`.
Strange: even with unicode literals it works fine for me on Python 2.7. I am on a Mac. Maybe only Windows has a problem with this.

But I do expect a standard string here, so forcing it to a string like you do is certainly good.

Reply:
I see a similar report here: https://github.com/peritus/bumpversion/issues/135 and here: https://github.com/theskumar/python-dotenv/pull/101

Very weird that it didn't occur earlier?